File: 0-100 LeetCode/1.py
def twoSum(self, nums: list[int], target: int) -> list[int]:
    hashmap = {}
    i = 0
    while i < len(nums):
        diff = target - nums[i]
        if diff in hashmap:
            return [hashmap[diff], i]
        hashmap[nums[i]] = i
        i += 1

File: 0-100 LeetCode/1.go
package alg

func twoSum(nums []int, target int) []int {
	hm := map[int]int{}

	i := 0
	for i < len(nums) {
		d := target - nums[i]
		if _, ok := hm[d]; ok {
			return []int{hm[d], i}
		}

		hm[nums[i]] = i
		i += 1
	}

	return []int{0, 0}
}
